iroh-net/src/hp/netcheck.rs

```diff
@@ -1084,11 +744,8 @@ impl Actor {
                 to_remove.push(*t);
                 continue;
             }
-            for (region_id, d) in &pr.region_latency {
-                let bd = best_recent.entry(*region_id).or_insert(*d);
-                if d < bd {
-                    *bd = *d;
-                }
+            for (region_id, d) in pr.region_latency.iter() {
+                best_recent.update_region(region_id, d);
```

Comment:
could even make this a dedicated method, `best_recent.merge(&pr.region_latency)`.